docs/conditions.md

```diff
@@ -0,0 +1,150 @@
+
+| Condition        | Status           |
+| :------------- |:-------------:|
+| OperatorAvailable | False |
+| OperatorProgressing | True |
+| OperatorDegraded | False |
+| Meaning | Operator is deploying the component |
+
+## Readiness Probe
+With a standardized set of conditions, the HCO should report the health of the
+overall application back to OLM and the user.  This will be critial for sensitive
+operations like upgrade, because OLM needs to know it shouldn't replace an
+operator when it is in the middle of imporant work.
+
+See this [issue](https://github.com/operator-framework/operator-lifecycle-manager/issues/922) for why we only want to report a readiness probe on the HCO
+instead of on all component operators.
+
+## Reason
+`Reason` is _a one-word CamelCase reason for the condition's last transition_.
+
+We'll use a series of lifecycle inspired prefixes paired with postfixes to
+standardize values for `Reason`.
+
+|         | -Failed  | -Succeeded | -Invalid | -InProgress |
+| :------------- |:-------------:|:-----:|:-----:|:-----:|
+| Install- | InstallFailed | InstallSucceeded | InstallInvalid | InstallInProgress |
+| Upgrade- | UpgradeFailed | UpgradeSucceded | UpgradeInvalid | UpgradeInProgress |
+| Heal- | HealFailed | HealSucceeded | HealInvalid | HealInProgress |
+| Configuration- | ConfigurationFailed | ConfigurationSucceeded | ConfigurationInvalid | ConfigurationInProgress |
+
+|         | -Failed  | -Succeeded | -Invalid | -InProgress |
+| :------------- |:-------------:|:-----:|:-----:|:-----:|
+| Meaning | The attempted operation **Failed** and the error is clear to the operator | The attempted operation **Succeeded** |  The attempted operation is missing something or is **Invalid** at this time | The attempted operation is **InProgress** |
+
+## Message
+`Message` is a _human-readable message indicating details about last transition_.
+
+Explain why your CR has `Reason`.
+
+
+## HCO conditions
+It's important to point out that the `ConditionTypes` on the HCO don't represent
+the condition the HCO is in, rather the condition of the component CRs.
+
+The HCO will use the same `ConditionType`s and `Reason`s, but it will be the
+only operator to use `Unknown` for `Status`.  If the HCO notices that a component
+CR is missing condition fields, the HCO will assume the status of
+`OperatorAvailable = false`, `OperatorProgressing = unknown` and
+`OperatorDegraded = unknown`.  If the HCO also detects there is no `Reason` value
+too, then it will assume `Reason = "InstallInvalid"`.
+
+The HCO's `ConditionType`s will always represent the _worst_ `Status` and the
+_most recently viewed_ `Reason`.
+
+The _worst_ `Status` for each `ConditionType`:
+
+| Condition   | Status  |
+| :------------- |:-------------:|
+| OperatorAvailable | False |
+| OperatorProgressing | False |
```

Comment:
I think that if any operator is Progressing...then the HCO is progressing.

Reply:
Do you think this should be renamed on the HCO?

Reply:
This was more to point out that when you say "worst" for a given condition type...that doens't really work for the OperatorProgressing (or ApplicationProgressing). Since, what we want to communicate, is that if any of the components are progressing, then the HCO is also progressing.

This wasn't really about renaming anything. Just that the "worst" for progressing should be "True"

Reply:
I agree.  If Progressing is shown on the HCO as false and one operator has "True", it should be shown as "True" or that will be confusing.

"Worst" doesn't apply uniformly for all the conditions because the worst case is that we're failed and we're not doing anything about it.  OperatorProgressing is really an outlier here.